### ISSUE TITLE
Array value in the model.

### ISSUE THREAD

**Issue by Dragomir-Ivanov** [casbin/casbin#187]
Hi there, I know I can configure policy like that in the model.

```ini
[policy_definition]
p = sub, obj, act
```

However `sub, obj and act` are always `string`. However I may have use-case where I would like to allow only access to certain fields enumerated in an `array`, along with custom function in the matcher.
Something like this

```ini
[request_definition]
r = sub, obj, act, flds

[policy_definition]
p = sub, obj, act, flds
```

```csv
p, alice, data1, read, (id, name, age)
```

where `data1` may have other fields, but `alice` is restricted to query only those.

I presume this is not supported, so are you willing PR for those?

**Comment by hsluoyz** [comment]
I understood your scenario. But your example is still not solid. For example, ``flds`` is an array (I think you mean Golang's slice?). But you mentioned ``(id, name, age)`` in the policy, so how is ``(id, name, age)`` mapped to the slice index?

**Comment by Dragomir-Ivanov** [comment]
First, thanks for the fast response! I don't quite understand your question about the `mapped slice index`, but I have build a test example. You can't find it here: https://repl.it/@Dragomir_Ivanov/EnviousMarriedSquares
It doesn't execute because the IDE doesn't support adding Go packages, but all files are there.
The important part is that I can't do `printFields(p.flds)`, because `p.flds` is a `string` and not `[]string`.

**However** I guess I can try parse the string coming from `casbin` in my own custom functions and do the logic there, but this ignores already what is in `govaluate`.
In conclusion I do think that `casbin` should support array of strings in its models as well.

**Comment by hsluoyz** [comment]
Your example is supported. See the RESTful example here: https://github.com/casbin/casbin#examples

```csv
p, alice, /alice_data/*, GET
p, alice, /alice_data/resource1, POST

p, bob, /alice_data/resource2, GET
p, bob, /bob_data/*, POST

p, cathy, /cathy_data, (GET)|(POST)
```

**Comment by hsluoyz** [comment]
A single field in a policy rule (e.g., CSV) cannot be interpreted as ``[]string``. It's not trivial for Casbin to support it natively. But you always have workarounds. For example, you can use the regex as I said before: ``(GET)|(POST)`` to contain two values. Or define your custom array like ``a $ b $ c``, which means an array of ``(a, b, c)``. Then you parse this string into real Golang slice in your custom function.

**Comment by Dragomir-Ivanov** [comment]
Thanks, I think regex way will somewhat work, albeit confusing at first.
Why it is not trivial for Casbin to support array of strings? Is it just CSV parsing, or something related to Casbin implementation ( I haven't read the source yet ). If CSV is the cause, maybe you can introduce another file format for policy definition, say JSON. Are you willing to support array of strings, even if we introduce special array syntax, say `p, cathy, /cathy_data, read|update`?
I refrain from supporting arrays in my custom functions, because I would like to use the arrays in the `govaluate` expressions ( `in` operator ).

**Comment by Dragomir-Ivanov** [comment]
Okay, I understand now.
Well, you can always have slice of empty interfaces, and put whatever you want in there.
Thanks for the explanation!
